script/pr-check.js

```diff
@@ -0,0 +1,190 @@
+
+  octokit.pulls.createReviewRequest({
+    ...octokitDefaults,
+    team_reviewers: ['frontend-review-group'],
+  });
+
+  return octokit.pulls.createReview({
+    ...octokitDefaults,
+    body: OVERALL_REVIEW_COMMENT,
+    event: 'COMMENT',
+    comments: additions.map(({ path, position }) => ({
+      path,
+      position,
+      body: LINE_COMMENT,
+    })),
+  });
+}
+
+/**
+ * Log the message & data, then exit
+ */
+function finish(message, data) {
+  console.log(message);
+  console.log(data);
+  process.exit();
+}
+
+/**
+ * Add a label to the beginning of added lines in the diff
+ * where the label consists of:
+ *   - filename
+ *   - offset from the top of the diff chunk
+ * And both of the label pieces are separated by a colon
+ *
+ * Modelled after a bash function from this SO answer:
+ * https://stackoverflow.com/a/12179492
+ */
+function labelAdditions(diffLines) {
+  let path = null;
+  let position = null;
+  let inNewFile = false;
+  const output = [];
+
+  diffLines.forEach(line => {
+    if (/--- (a\/)?.*/.test(line)) {
+      return;
+    }
+    const match = line.match(/\+\+\+ (b\/)?(.*$)/);
+    if (match) {
+      path = match[2];
+    } else if (inNewFile && /@@ .* @@.*/.test(line)) {
+      // Only reset the position if our current line is the beginning of a diff chunk
+      // AND if the previous line marked the beginning of a new file
+      position = 1;
+    } else if (/@@ -[0-9]+(,[0-9]+)? \+([0-9]+)(,[0-9]+)? @@.*/.test(line)) {
+      // Increment the position when we reach the beginning of a new diff chunk
+      position++;
+    } else if (/^([ +-]).*/.test(line)) {
+      // Only add to the output if this line of content is an addition (begins with a "+")
+      if (/^[+].*/.test(line)) output.push({ path, position, line });
+      // Increment position for each line of actual content in the diff
+      position++;
+    }
+    inNewFile = !!match;
+  });
+  return output;
+}
+
+/**
+ * Tries to filter the array down to items which contain the CODE_PATTERN
+ * Throws a Promise rejection if the filtered list is empty
+ */
+function findPattern(arr) {
+  return Promise.resolve(
+    arr.filter(({ line }) => new RegExp(CODE_PATTERN).test(line)),
+  ).then(
+    patternList =>
+      isEmpty(patternList)
+        ? finish(
+            `No additions found matching the pattern: "${CODE_PATTERN}"`,
+            patternList,
+          )
+        : patternList,
+  );
+}
+
+/**
+ * Receives a list of additions and returns a list containing those additions which (if any)
+ * area considered "new" in this run.
+ */
+function filterAgainstPreviousComments(additions) {
+  return getPRbotComments()
+    .then(comments =>
+      // Create sorted list of unique stringified objects for easy comparison
+      // Removing the duplicates shouldn't be necessary once everything is working properly
+      // because there shouldn't be any duplicates
+      sortBy(uniqBy(comments, JSON.stringify), JSON.stringify).map(
```

Comment:
Would it be better to reverse the operations (`sortBy` and then `uniqBy` or [`sortedUniqBy`](https://lodash.com/docs/4.17.15#sortedUniqBy))?